spec/models/armor_spec.rb

```diff
@@ -156,10 +120,85 @@
       let(:armor) { build(:armor) }
 
       it 'is invalid' do
-        armor.validate
+        validate
         expect(armor.errors[:base]).to include "doesn't match an armor item that exists in Skyrim"
       end
     end
+
+    context 'when updating in-game item attributes' do
+      let(:armor) { create(:armor, :with_matching_canonical) }
+
+      context 'when the update changes the canonical association' do
+        let!(:new_canonical) do
+          create(
+            :canonical_armor,
+            name: 'Imperial Boots of Resist Frost',
+            weight: 'light armor',
+            magical_effects: 'This Will Be Case Insensitive',
+            unit_weight: 2,
+          )
+        end
+
+        it 'changes the canonical association' do
+          armor.name = 'Imperial boots of resist frost'
+          armor.magical_effects = 'this will be case insensitive'
+          armor.weight = nil
+          armor.unit_weight = nil
+
+          expect { validate }
+            .to change(armor, :canonical_armor)
+                  .to(new_canonical)
+        end
+
+        it 'sets attributes on the in-game item', :aggregate_failures do
+          armor.name = 'Imperial boots of resist frost'
+          armor.magical_effects = 'this will be case insensitive'
+          armor.weight = nil
+          armor.unit_weight = nil
+
+          validate
+
+          expect(armor.name).to eq 'Imperial Boots of Resist Frost'
+          expect(armor.magical_effects).to eq 'This Will Be Case Insensitive'
+          expect(armor.weight).to eq 'light armor'
+          expect(armor.unit_weight).to eq 2
+        end
+      end
+
+      context 'when the update results in an ambiguous match' do
```

Comment:
Let's add a test to this context to ensure attributes are not set from ambiguous canonical matches.